src/v/security/audit/schemas/types.h

```diff
@@ -0,0 +1,481 @@
+#include "utils/named_type.h"
+#include "version.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <string_view>
+
+namespace security::audit {
+static constexpr std::string_view ocsf_api_version = "1.0.0";
+static constexpr std::string_view vendor_name = "Redpanda Data, Inc.";
+
+using port_t = named_type<uint16_t, struct port_t_type>;
+using timestamp_t = named_type<long, struct timestamp_t_type>;
+using type_uid = named_type<int, struct type_uid_type>;
+
+template<typename T>
+concept has_equality_fields = requires(T t) { t.equality_fields(); };
+
+enum class category_uid : int {
+    system_activity = 1,
+    findings = 2,
+    iam = 3,
+    network_activity = 4,
+    discovery = 5,
+    application_activity = 6
+};
+
+enum class class_uid : int {
+    file_system_activity = 1001,
+    kernel_extension_activity = 1002,
+    kernel_activity = 1003,
+    memory_activity = 1004,
+    module_activity = 1005,
+    scheduled_job_activity = 1006,
+    process_activity = 1007,
+    security_finding = 2001,
+    account_change = 3001,
+    authentication = 3002,
+    authorize_session = 3003,
+    entity_management = 3004,
+    user_access_management = 3005,
+    group_management = 3006,
+    network_activity = 4001,
+    http_activity = 4002,
+    dns_activity = 4003,
+    dhcp_activity = 4004,
+    rdp_activity = 4005,
+    smb_activity = 4006,
+    ssh_activity = 4007,
+    ftp_activity = 4008,
+    email_activity = 4009,
+    network_file_activity = 4010,
+    email_file_activity = 4011,
+    email_url_activity = 4012,
+    device_inventory_info = 5001,
+    device_config_state = 5002,
+    web_resource_activity = 6001,
+    application_lifecycle = 6002,
+    api_activity = 6003,
+    web_resource_access_activity = 6004
+};
+
+enum class severity_id : int {
+    unknown = 0,
+    informational = 1,
+    low = 2,
+    medium = 3,
+    high = 4,
+    critical = 5,
+    fatal = 6,
+    other = 99,
+};
+
+struct api {
+    ss::sstring operation;
+
+    auto equality_fields() const { return std::tie(operation); }
+};
+
+struct product {
+    ss::sstring name;
+    ss::sstring vendor_name;
+    ss::sstring version;
+
+    auto equality_fields() const {
+        return std::tie(name, vendor_name, version);
+    }
+};
+
+static inline const product& redpanda_product() {
+    static const product redpanda_product_struct{
+      .name = "Redpanda",
+      .vendor_name = ss::sstring{vendor_name},
+      .version = ss::sstring{redpanda_git_version()}};
+    return redpanda_product_struct;
+}
+
+struct metadata {
+    product product;
+    ss::sstring version;
+
+    auto equality_fields() const { return std::tie(product, version); }
+};
+
+static inline const metadata& ocsf_redpanda_metadata() {
+    static const metadata ocsf_metadata{
+      .product = redpanda_product(), .version = ss::sstring{ocsf_api_version}};
+
+    return ocsf_metadata;
+}
+
+struct network_endpoint {
+    fragmented_vector<ss::sstring> intermediate_ips;
```

Comment:
how big would this ever get? So big that we'd need to worry about fragmentation? I'd think a normal vector is good here

Reply:
same for other vectors defined in this header

Reply:
Sure, whatever you think